src/main/java/com/dabsquared/gitlabjenkins/trigger/handler/merge/MergeRequestHookTriggerHandlerImpl.java

```diff
@@ -198,6 +198,7 @@ protected CauseData retrieveCauseData(MergeRequestHook hook) {
                 .withSourceBranch(hook.getObjectAttributes().getSourceBranch())
                 .withUserName(
                         hook.getObjectAttributes().getLastCommit().getAuthor().getName())
+                .withUserUsername(hook.getUser().getUsername())
```

Comment:
If the `UserUsername` cannot be returned here with `hook.getObjectAttributes().getLastCommit().getAuthor().getUsername()` then we will need some upstream changes. I don't think we are supposed to stray from the pattern prescribed, as the suggested `hook.getUser().getUsername()` would obviously return something not related to the Last Commit Author.

Reply:
Yeah, and I mentioned that it can be confusing that **gitlabUserName** and **gitlabUserUsername** can represent two different persons.
It was probably a mistake that the author of the last commit was set as the author of the merge request (in case of MR Event) or comment (in case of Note Event).